src/Cli/ConfigGenerator.cs

```diff
@@ -535,6 +541,98 @@ public static bool TryConfigureSettings(ConfigureOptions options, FileSystemRunt
+        /// </summary>
+        /// <param name="options">The configuration options provided by the user.</param>
+        /// <param name="runtimeConfig">The runtime configuration to be updated. This parameter is passed by reference and must not be null if the method returns true.</param>
+        /// <returns>
+        /// True if the data source options were successfully configured and the runtime configuration was updated; otherwise, false.
+        /// </returns>
+        private static bool TryConfigureDataSourceOptions(
+            ConfigureOptions options,
+            [NotNullWhen(true)] ref RuntimeConfig runtimeConfig)
+        {
+            DatabaseType dbType = runtimeConfig.DataSource.DatabaseType;
+            string dataSourceConnectionString = runtimeConfig.DataSource.ConnectionString;
+
+            if (options.DataSourceDatabaseType is not null)
+            {
+                if (!Enum.TryParse(options.DataSourceDatabaseType, ignoreCase: true, out dbType))
+                {
+                    _logger.LogError(EnumExtensions.GenerateMessageForInvalidInput<DatabaseType>(options.DataSourceDatabaseType));
+                    return false;
+                }
+            }
+
+            if (options.DataSourceConnectionString is not null)
+            {
+                dataSourceConnectionString = options.DataSourceConnectionString;
+            }
+
+            Dictionary<string, object?>? dbOptions = new();
+            HyphenatedNamingPolicy namingPolicy = new();
+
+            if (DatabaseType.CosmosDB_NoSQL.Equals(dbType))
+            {
+                AddCosmosDbOptions(dbOptions, options, namingPolicy);
+            }
+            else if (!string.IsNullOrWhiteSpace(options.DataSourceOptionsDatabase)
+                    || !string.IsNullOrWhiteSpace(options.DataSourceOptionsContainer)
+                    || !string.IsNullOrWhiteSpace(options.DataSourceOptionsSchema))
+            {
+                _logger.LogError("Database, Container, and Schema options are only applicable for CosmosDB_NoSQL database type.");
+                return false;
+            }
+
+            if (options.DataSourceOptionsSetSessionContext is not null)
+            {
+                if (!(DatabaseType.MSSQL.Equals(dbType) || DatabaseType.DWSQL.Equals(dbType)))
+                {
+                    _logger.LogError("SetSessionContext option is only applicable for MSSQL/DWSQL database type.");
+                    return false;
+                }
+
+                dbOptions.Add(namingPolicy.ConvertName(nameof(MsSqlOptions.SetSessionContext)), options.DataSourceOptionsSetSessionContext.Value);
+            }
+
+            dbOptions = dbOptions.IsNullOrEmpty() ? null : dbOptions;
+            DataSource dataSource = new(dbType, dataSourceConnectionString, dbOptions);
+            runtimeConfig = runtimeConfig with { DataSource = dataSource };
+
+            return true;
+        }
```

Comment:
Should returning true be conditioned on runtimeconfig != null to honor the promise you make in the function signature and description?

Reply:
sure.